server/services/v1/database/search_indexer.go

```diff
@@ -60,7 +60,8 @@ func (i *SearchIndexer) OnPostCommit(ctx context.Context, tenant *metadata.Tenan
 		}
 
 		collection := db.GetCollection(collName)
-		if collection == nil {
+		if collection == nil || event.Key == nil {
```

Comment:
We should not see event from drop table here. Are you events for delete with empty filters?

Reply:
Check StatsTxStore.DropTable I am using Delete to transactionally drop statistics along with the data.

Reply:
Okay so the stats table is part of user subspace that's why it is coming to the indexer and I am assuming all the stats operations are not pushed to listener because they are using some other APIs.